### PR TITLE
added cart amount validation

### PR DESCRIPTION
**Pull request by abdulwahidsharief**
creates a new razorpay order if cart amount mismatches

### CODE REVIEW

File: catalog/controller/extension/payment/razorpay.php [hunk 1]
```diff
@@ -103,10 +103,17 @@ public function index()
                 // Orders API with payment autocapture
                 $order_data = $this->get_order_creation_data($this->session->data['order_id']);
 
-                if(empty($this->session->data["razorpay_order_id_" . $this->session->data['order_id']]) === true)
+                if(isset($this->session->data["razorpay_order_amount"]) === false)
```
Comment (enishant) [inline catalog/controller/extension/payment/razorpay.php:106]: @abdulwahidsharief please add space after `if`

File: catalog/controller/extension/payment/razorpay.php [hunk 2]
```diff
@@ -103,10 +103,19 @@ public function index()
...
 
-                if(empty($this->session->data["razorpay_order_id_" . $this->session->data['order_id']]) === true)
+                if (isset($this->session->data["razorpay_order_amount"]) === false)
+                {
+                    $this->session->data["razorpay_order_amount"] = 0;
+                }
+
+                if ((isset($this->session->data["razorpay_order_id_" . $this->session->data['order_id']]) === false) or
+                    (isset($this->session->data["razorpay_order_id_" . $this->session->data['order_id']]) === true and
+                        ($this->session->data["razorpay_order_amount"] === 0 or
```
Comment (nikhilsharma1100) [inline catalog/controller/extension/payment/razorpay.php:113]: Please align all conditional statement under each other in this if block

Comment (abdulwahidsharief) [reply catalog/controller/extension/payment/razorpay.php:113]: @nikhilsharma1100 every condition should start with new line.

Comment (ramth05) [reply catalog/controller/extension/payment/razorpay.php:113]: @abdulwahidsharief , please use bracket for each condition for ex. 
`($this->session->data["razorpay_order_amount"] === 0)`

Comment (abdulwahidsharief) [reply catalog/controller/extension/payment/razorpay.php]: @ramth05 done

**Review by nikhilsharma1100 — APPROVED** [review]

**Review by nikhilsharma1100 — COMMENTED** [review]
LGTM

**Pull request merged**